service/controller/resource/deployment/deployment.go

```diff
@@ -2,36 +2,40 @@ package deployment
 
 import (
 	"context"
+	"net"
 
 	azureresource "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-05-01/resources"
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/microerror"
 
-	"github.com/giantswarm/azure-operator/v4/service/controller/controllercontext"
 	"github.com/giantswarm/azure-operator/v4/service/controller/key"
 	"github.com/giantswarm/azure-operator/v4/service/controller/resource/deployment/template"
+	"github.com/giantswarm/azure-operator/v4/service/network"
 )
 
 func (r Resource) newDeployment(ctx context.Context, customObject providerv1alpha1.AzureConfig, overwrites map[string]interface{}) (azureresource.Deployment, error) {
-	cc, err := controllercontext.FromContext(ctx)
+	// The VPN subnet is not persisted in the AzureConfig so I have to compute it now.
```

Comment:
Should we address this or should we wait until ClusterAPI? @tuommaki

Reply:
I don't fully get how this is sub-optimal? When we can compute something like this, I think we should. The computation should be deterministic and cheap by default.

Reply:
@tuommaki the VPN subnet calculation is suboptimal because it is different from the other subnets.
The other ones are calculated once and stored in the CR, the VPN one is calculated all the times.
We should add it to the CR, but since it will be unuseful soon that is not a smart thing to do IMHO.

WDYT?

Reply:
When it's about to be dropped soon anyway, just leave it like that.

Reply:
agreed